app/models/gem_typo.rb

```diff
@@ -3,43 +3,33 @@
 class GemTypo
   attr_reader :protected_gem
 
-  include Gem::Text
-
-  DOWNLOADS_THRESHOLD = 10_000_000
-  SIZE_THRESHOLD = 4
-
   def initialize(rubygem_name)
-    @rubygem_name       = rubygem_name.downcase
-    @distance_threshold = distance_threshold
+    @rubygem_name = rubygem_name
   end
 
   def protected_typo?
-    return false if @rubygem_name.size < GemTypo::SIZE_THRESHOLD
+    return false if GemTypoException.where("upper(name) = upper(?)", @rubygem_name).any?
 
-    gem_typo_exceptions = GemTypoException.all.pluck(:name)
-    return false if gem_typo_exceptions.include?(@rubygem_name)
+    return false if published_exact_name_matches.any?
 
-    protected_gems.each do |protected_gem|
-      distance = levenshtein_distance(@rubygem_name, protected_gem)
-      if distance <= @distance_threshold
-        @protected_gem = protected_gem
-        return true
-      end
-    end
+    match = matched_protected_gem_names.select(:id, :name).first
```

Comment:
why are we selecting `id` here?

Reply:
> why are we selecting `id` here?

For backwards compatibility with any existing code that was using the match to render an error message. It selected the object, so this selects the ID and the name as a minimum set needed to instantiate the model for that purpose.

Reply:
> For backwards compatibility with any existing code

I don't think there is any existing code that uses id. If there is such code can you please point it out to me? My issue with using ID here is that it will create confusion in the future about why was this added if not used anywhere.